client/models.py

```diff
@@ -0,0 +1,51 @@
+from django.db import models
+from django.core.validators import MinLengthValidator
+from django.core.exceptions import ValidationError
+from buy_together_app.models import User
+
+
+class Client(User):
+    area = models.CharField(max_length=32,
+                            validators=[MinLengthValidator(1)],
+                            help_text="area")
+
+    @staticmethod
+    def get_all_clients():
+        """Fetches list of all Clients.
+        Args:
+            None.
+        Returns:
+            returns a queryset containing all the Clients.
+        """
+        return Client.objects.all()
+
+    @staticmethod
+    def save_client(client):
+        """Saves new client 'safely'
+        Args:
+            client - the Client wanted to be saved.
+        Returns:
+            None.
+        raises:
+        ValidationError error: if fields input aren't valid.
+        """
+        try:
+            Client.full_clean(client)
+            client.save()
+        except ValidationError as error:
+            raise error
+
+    @staticmethod
+    def delete_client(client):
+        """Deletes client 'safely'
+        Args:
+            client - the client wanted to be deleted
+        Returns:
+            None.
+        raises:
+        ValueError error: if the client is not in DB.
+        """
+        try:
+            client.delete()
+        except ValueError as error:
```

Comment:
1. does delete raise a ValueError?
2. why do we catch the exception if the only thing we're doing with it is raising it again? there is no special handling made (no message to the user, no raising a different error... ) please look at the comments I left for Paz regarding the same subject:  https://github.com/redhat-beyond/BuyTogether/pull/16#discussion_r1049445757

Reply:
Ok I understood and agree. 
i thought that if I try to delete client that already deleted so his value will be 'None'.. this is why i used ValueError. 

I changed it   👍